src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -3572,7 +3574,10 @@ private int index(Rope source, Rope other, int byteOffset, Encoding enc, Normali
             int p = 0;
             int end = p + source.byteLength();
             if (byteOffset != 0) {
-                byteOffset = source.isSingleByteOptimizable() ? byteOffset : StringSupport.offset(enc, bytes, p, end, byteOffset);
+                if (!source.isSingleByteOptimizable()) {
+                    final int pp = byteIndexFromCharIndexNode.execute(source, p, byteOffset);
+                    byteOffset = StringSupport.offset(p, end, pp);
```

Comment:
What is `pp`?

Reply:
This was inlined from `StringSupport.offset`. `p` is supposed to be a pointer. `pp` is just a second pointer from the second call to `byteIndexFromCharIndex`. I need to go through this primitive again, but I really wasn't touching it for this PR. It only got affected because I wanted to delete that code from `StringSupport`.

Reply:
It would be nice to have some JavaDoc for those non-obvious StringSupport methods. Doesn't need to be in this PR.

Reply:
`p` could also be replaced by `0` on this line.